plugins/hydra_ray_launcher/hydra_plugins/hydra_ray_launcher/conf/__init__.py

```diff
@@ -0,0 +1,199 @@
+    ray_remote_cfg: Dict[str, Any] = field(
+        default_factory=lambda: {"num_cpus": 1, "num_gpus": 0}
+    )
+
+
+# Ray AWS config
+
+
+class RayAutoScalingMode(Enum):
+    default = "default"
+    aggressive = "aggressive"
+
+
+@dataclass
+class RayDockerConf:
+    """
+    This executes all commands on all nodes in the docker container,
+    and opens all the necessary ports to support the Ray cluster.
+    Empty string means disabled.
+
+    """
+
+    # e.g., tensorflow/tensorflow:1.5.0-py3
+    image: str = ""
+    # e.g. ray_docker
+    container_name: str = ""
+    # If true, pulls latest version of image. Otherwise, `docker run` will only pull the image
+    # if no cached version is present.
+    pull_before_run: bool = True
+    # Extra options to pass into "docker run"
+    run_options: List[str] = field(default_factory=list)
+
+
+@dataclass
+class RayProviderConf:
+    type: str = "aws"
+    region: str = "us-west-2"
+    availability_zone: str = "us-west-2a,us-west-2b"
+    cache_stopped_nodes: bool = False
+    key_pair: Dict[str, str] = field(default_factory=lambda: {"key_name": "hydra"})
+
+
+@dataclass
+class RsyncConf:
+    """
+    If your code contains more than one modules/files, you need to sync your code to the remote cluster.
+    The code will be synced to a tmp dir on the cluster.
+    code_dir could be a relative dir to where you run your app or a absolute dir.
+    Leave this empty if you don't need to sync code.
+
+    The conf is equivalent to run:
+    rsync {source} {target} --include={include} --exclude={exclude}
+
+    source/target will either be remote/local or local/remote depending on sync_up or sync down
+    Read RayAWSConf for more details.
+    """
+
+    source_dir: Optional[str] = None
+    target_dir: Optional[str] = None
+    include: List[str] = field(default_factory=list)
+    exclude: List[str] = field(default_factory=list)
+
+
+@dataclass
+class RayClusterConf:
+    """
+    This class maps to a Ray cluster config, e.g:
+    https://github.com/ray-project/ray/blob/master/python/ray/autoscaler/aws/example-full.yaml
+    Schema: https://github.com/ray-project/ray/blob/master/python/ray/autoscaler/ray-schema.json
+    """
+
+    # An unique identifier for the head node and workers of this cluster.
+    cluster_name: str = "default"
+
+    # The minimum number of workers nodes to launch in addition to the head
+    # node. This number should be >= 0.
+    min_workers: int = 0
+
+    # The maximum number of workers nodes to launch in addition to the head
+    # node. This takes precedence over min_workers.
+    max_workers: int = 1
+
+    # The initial number of worker nodes to launch in addition to the head
+    # node. When the cluster is first brought up (or when it is refreshed with a
+    # subsequent `ray up`) this number of nodes will be started.
+    initial_workers: int = 0
+
+    autoscaling_mode: RayAutoScalingMode = RayAutoScalingMode.default
```

Comment:
can you add a comment about autoscaling_mode?

Reply:
The ray documentation doesn't really have much details on this. This is the field in their (schema)[https://github.com/ray-project/ray/blob/0d5b09f42646592222c5c4ade333ee9bb8807a6e/python/ray/autoscaler/ray-schema.json#L41-L45]

I'm not sure how to add a more useful comment so instead I added ray's schema info there with the RayAWSConfig.